### PR TITLE
Model creation#9

### PR DESCRIPTION
**Pull request by david-ron**
#9 adding generic models

### CODE REVIEW

**Comment by david-ron** [comment]
Might still be missing some attributes

File: src/main/java/com/kimsquitsystem/kimsquitdemo/model/Announcement.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
+package com.kimsquitsystem.kimsquitdemo.model;
+
+public class Announcement {
+    private String announcementId;
```
Comment (bit172) [inline src/main/java/com/kimsquitsystem/kimsquitdemo/model/Announcement.java]: These variables need to be final and setters need to be removed.

Comment (bit172) [reply src/main/java/com/kimsquitsystem/kimsquitdemo/model/Announcement.java]: Same with other models

File: src/main/java/com/kimsquitsystem/kimsquitdemo/model/Course.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
+package com.kimsquitsystem.kimsquitdemo.model;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class Course {
+    private String courseName;
+    private String courseID;
+    // For userIDs to connect to User object which may be a student or teacher
+    private HashMap<String,User> membersofCourse;
```
Comment (bit172) [inline src/main/java/com/kimsquitsystem/kimsquitdemo/model/Course.java]: Why is this a hash map?

**Review by leojrsilao — APPROVED** [review]
Looks good! Some attributes can be modified tho BET!!!!

File: src/main/java/com/kimsquitsystem/kimsquitdemo/model/User.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
+package com.kimsquitsystem.kimsquitdemo.model;
+
+import java.util.HashMap;
+
+public class User {
```
Comment (bit172) [inline src/main/java/com/kimsquitsystem/kimsquitdemo/model/User.java:5]: We need to discuss about having inherited users or just Teachers and Students

**Review by bit172 — CHANGES_REQUESTED** [review]

File: src/main/java/com/kimsquitsystem/kimsquitdemo/model/Course.java (hunk 2 above)
Comment (david-ron) [reply src/main/java/com/kimsquitsystem/kimsquitdemo/model/Course.java]: it has been removed

File: src/main/java/com/kimsquitsystem/kimsquitdemo/model/User.java (hunk 3 above)
Comment (david-ron) [reply src/main/java/com/kimsquitsystem/kimsquitdemo/model/User.java:5]: students and teachers share a lot of the same attributes

File: src/main/java/com/kimsquitsystem/kimsquitdemo/model/Announcement.java (hunk 1 above)
Comment (david-ron) [reply src/main/java/com/kimsquitsystem/kimsquitdemo/model/Announcement.java]: okk

File: src/main/java/com/kimsquitsystem/kimsquitdemo/model/User.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+     */
+    public String getLastName() {
+        return lastName;
+    }
+
+    /**
+     * get User Id
+     * @return
+     */
+    final public String getUserId() {
```
Comment (bit172) [inline src/main/java/com/kimsquitsystem/kimsquitdemo/model/User.java]: not final

File: src/main/java/com/kimsquitsystem/kimsquitdemo/model/Course.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+     */
+    final public String getCourseName() {
+        return courseName;
+    }
+
+    /**
+     * get course id
+     * @return
+     */
+    final public String getCourseID() {
```
Comment (bit172) [inline src/main/java/com/kimsquitsystem/kimsquitdemo/model/Course.java]: not final

File: src/main/java/com/kimsquitsystem/kimsquitdemo/model/Course.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        this.courseName = courseName;
+        this.courseID = courseId;
+        this.announcements = announcements;
+    }
+
+    /**
+     * get name of course
+     * @return
+     */
+    final public String getCourseName() {
```
Comment (bit172) [inline src/main/java/com/kimsquitsystem/kimsquitdemo/model/Course.java]: not final

**Review by bit172 — CHANGES_REQUESTED** [review]
Do the changes and I'll merge. Need a meeting to get things clear

**Comment by david-ron** [comment]
> Do the changes and I'll merge. Need a meeting to get things clear

changes made

**Review by bit172 — APPROVED** [review]

**Pull request merged**